src/include/duckdb/catalog/catalog_entry/type_catalog_entry.hpp

```diff
@@ -0,0 +1,35 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/catalog/catalog_entry/type_catalog_entry.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/catalog/standard_entry.hpp"
+#include "duckdb/common/mutex.hpp"
+#include "duckdb/parser/parsed_data/create_type_info.hpp"
+
+namespace duckdb {
+class Serializer;
+class Deserializer;
+
+//! A type catalog entry
+class TypeCatalogEntry : public StandardEntry {
+public:
+	//! Create a TypeCatalogEntry and initialize storage for it
+	TypeCatalogEntry(Catalog *catalog, SchemaCatalogEntry *schema, CreateTypeInfo *info);
+
+	shared_ptr<LogicalType> user_type;
```

Comment:
This doesn't need to be a shared_ptr: the heavy parts of the LogicalType (i.e. the `ExtraTypeInfo`) are already a `shared_ptr` internally.